build_tools/benchmarks/benchmark_helper.py

```diff
@@ -141,12 +146,61 @@ def _dump_cmds_handler(e2e_test_artifacts_dir: pathlib.Path,
   print(*lines, sep="\n")
 
 
+def merge_results(benchmark_results_files: List[io.TextIOWrapper]):
+  # Merges the benchmark results from `right` into `left` and returns the updated `left`
+  def _merge_two_resultsets(left, right):
+    # Parses a JSON benchmark results file and makes some sanity checks
+    def _load(file):
+      try:
+        content = json.load(file)
+      except json.JSONDecodeError as e:
+        raise RuntimeError('"{}" seems not to be a valid JSON file: {}'.format(
+            file.name, e.msg))
```

Comment:
If the `file` here is `io.TextIOWrapper`, it looks like they don't actually have the `name` field? (at least according to doc https://docs.python.org/3/library/io.html#io.TextIOWrapper)

Reply:
I can see this might be the only way to get filenames for error messages from `argparse.FileType`.

Alternatively, we can use `pathlib.Path` on the argument type and open file by ourselves here (so we still have the original path for the file name). But this does make testing harder, because now you need to create temp files instead of using `StringIO` as inputs to test it?

Either way seems okay to me, I'm just a little worried about using undocumented fields.